src/main/java/appeng/fluids/container/ContainerFluidInterface.java

```diff
@@ -67,56 +67,49 @@ public void detectAndSendChanges()
 
 		if( Platform.isServer() )
 		{
-			this.sendTankUpdate();
+			this.tankSync.sendDiff( this.listeners );
 		}
 
 		super.detectAndSendChanges();
 	}
 
 	@Override
-	public void addListener( IContainerListener listener )
+	protected void setupConfig()
 	{
-		super.addListener( listener );
-		this.sendTankInfo( listener );
 	}
 
-	private void sendTankUpdate( final IContainerListener l, final HashMap<Integer, NBTTagCompound> updateMap )
+	@Override
+	protected void loadSettingsFromHost( final IConfigManager cm )
 	{
-		if( l instanceof EntityPlayerMP )
-		{
-			NetworkHandler.instance().sendTo( new PacketFluidTank( updateMap ), (EntityPlayerMP) l );
-		}
 	}
 
-	private void sendTankUpdate()
+	@Override
+	public void addListener( IContainerListener listener )
 	{
-		final HashMap<Integer, NBTTagCompound> updateMap = new HashMap<>();
-		if( this.myDuality.writeTankInfo( updateMap, false ) )
-		{
-			for( final IContainerListener listener : this.listeners )
-			{
-				this.sendTankUpdate( listener, updateMap );
-			}
-		}
+		super.addListener( listener );
+		this.tankSync.sendFull( Collections.singleton( listener ) );
 	}
 
-	private void sendTankInfo( final IContainerListener l )
+	@Override
+	public void receiveFluidSlots( Map<Integer, IAEFluidStack> fluids )
 	{
-		final HashMap<Integer, NBTTagCompound> updateMap = new HashMap<>();
-		if( this.myDuality.writeTankInfo( updateMap, true ) )
-		{
-			this.sendTankUpdate( l, updateMap );
-		}
+		super.receiveFluidSlots( fluids );
+		this.tankSync.readPacket( fluids );
+	}
+
+	protected boolean supportCapacity()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the "@Override" annotation above this method signature [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1161)